cleanlab_studio/studio/studio.py

```diff
@@ -63,21 +66,41 @@ def upload_dataset(
         dataset_name: Optional[str] = None,
         *,
         schema_overrides: Optional[List[SchemaOverride]] = None,
+        **kwargs: Any,
     ) -> str:
         """
         Uploads a dataset to Cleanlab Studio.
 
         Args:
             dataset: Object representing the dataset to upload. Currently supported formats include a `str` path to your dataset, a pandas, snowflake, or pyspark DataFrame.
             dataset_name: Name for your dataset in Cleanlab Studio (optional if uploading from filepath).
-            schema_overrides: Optional dictionary of overrides you would like to make to the schema of your dataset. If not provided, schema will be inferred. Format defined [here](/guide/concepts/datasets/#schema-overrides).
-            modality: Optional parameter to override the modality of your dataset. If not provided, modality will be inferred.
-            id_column: Optional parameter to override the ID column of your dataset. If not provided, a monotonically increasing ID column will be generated.
+            schema_overrides: Optional list of overrides you would like to make to the schema of your dataset. If not provided, all columns will be untyped. Format defined [here](/guide/concepts/datasets/#schema-overrides).
+            modality: [DEPRECATED] Optional parameter to override the modality of your dataset. If not provided, modality will be inferred.
+            id_column: [DEPRECATED] Optional parameter to override the ID column of your dataset. If not provided, a monotonically increasing ID column will be generated.
 
         Returns:
             ID of uploaded dataset.
         """
         ds = init_dataset_source(dataset, dataset_name)
+        if kwargs.get("modality") is not None:
+            warnings.warn(
+                "Ignoring `modality` parameter which is deprecated and will be removed in a future release.",
+                DeprecationWarning,
+            )
+        if kwargs.get("id_column") is not None:
+            warnings.warn(
+                "Ignoring `id_column` parameter which is deprecated and will be removed in a future release.",
+                DeprecationWarning,
+            )
+
+        if isinstance(schema_overrides, dict):
+            # TODO: link to documentation for schema override format
+            warnings.warn(
+                "Using deprecated `schema_overrides` format. Please use list of SchemaOverride objects instead.",
+                DeprecationWarning,
+            )
```

Comment:
should warn _after_ converting schema overrides to prevent case where we warn and then fail